modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentItemResourceImpl.java

```diff
@@ -145,14 +149,21 @@ public ShipmentItem patchShipmentItem(
 			Long shipmentItemId, ShipmentItem shipmentItem)
 		throws Exception {
 
+		CommerceInventoryWarehouse commerceInventoryWarehouse =
+			ShipmentItemUtil.getCommerceInventoryWarehouse(
+				contextCompany.getCompanyId(), shipmentItem,
+				_commerceInventoryWarehouseService);
+
 		CommerceShipmentItem commerceShipmentItem =
 			_commerceShipmentItemService.getCommerceShipmentItem(
 				shipmentItemId);
 
 		_commerceShipmentItemService.updateCommerceShipmentItem(
 			shipmentItemId,
 			GetterUtil.get(
-				shipmentItem.getWarehouseId(),
+				Long.valueOf(
+					commerceInventoryWarehouse.
+						getCommerceInventoryWarehouseId()),
```

Comment:
see the other comments on defaults... also in this case if we don't pass any value (no id or ERC) because it's a patch I think we are throwing error wrongly